added sloppy file test

malud opened the pull request.

Commit:
added sloppy file test

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,86 @@
+      cmd: sleep 20
+      dependencies:
+      - ../apps/db
+      - ../apps/wordpress
```
I don't think this dependency should be here (only ../apps/db) - sloppy.io dependencies are implemented by Marathon, which are transitive: https://mesosphere.github.io/marathon/docs/rest-api.html#dependencies-array-of-strings

malud replied:
This is a simple test for resolving `depends_on` and `links`

jzaefferer replied:
Right, but the input only has `depends_on: db`, not both.

malud replied:
and `links` ? there is wordpress declared. should it be ignored completely ?

jzaefferer replied:
Wasn't `links` deprecated anyway? If its safe to simply ignore it, I'd be fine with that.

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,86 @@
+      instances: 1
+      mem: 256
+      port: 80
+      port_mappings:
```
This is redundant to the `port` property. We have it in our json format, but not in the yml format.

malud replied:
`port_mappings` is there by accident

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,86 @@
+      - ../apps/db
+      domain: mywords.sloppy.zone
+      env:
+      - WORDPRESS_DB_HOST=db.apps.pkg:3306
```
Where is this mapping implemented? This? https://github.com/sloppyio/sloppose/blob/b272f2360aba544a4152105195fae1dc217c2c94/pkg/converter/linker.go#L71-L77

malud replied:
Yes but gets prepared by the buildLinks func (string Template)

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,86 @@
+      dependencies:
+      - ../apps/db
+      - ../apps/wordpress
+      domain: $URI
```
This line shouldn't be here. The input doesn't have a domain for this app.

jzaefferer commented on pkg/converter/testdata/docker-compose-v2.yml:
```diff
@@ -10,8 +10,7 @@ services:
       - db
   wordpress:
-   depends_on:
-     - db
```
Why did you remove this?

malud replied:
As seen on the busy_box service, services which are declared in `depends_on` and `links` getting also resolved. So to ensure that the dependencies gets set by the host:port search - I've removed this.

jzaefferer replied:
I don't quite understand that part.

Since you mention host:port search: Env vars could reference other hosts (from other services in the same project) without specifying a port. Is that covered anywhere?

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,86 @@
+      mem: 256
+      volumes:
+      - container_path: /var/lib/mysql
+        size: 8GB
```
We need a way to specify the volume size, but there is no need to add the default. The size property is optional and the default is filled in on our backend.

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,86 @@
+      - MYSQL_USER=wordpress
+      image: mysql:8.0.0"
+      instances: 1
+      mem: 256
```
Similar to volume size, instances and mem also have defaults, so there is no need to generate those.

Though it would be good to add memory to one of the services in the input file and match it here.

jzaefferer reviewed (changes requested):
Would be great to have a similar test for the v3 docker-compose file format.

Commit:
added coveralls test coverage report

Commit:
removing defaults for instance count, memory and volume sizes

added test for compose v3 file

added an additional linker test

added coverage badge

Commit:
added additional tests

jzaefferer commented on pkg/converter/sloppy_file_test.go:
```diff
@@ -0,0 +1,89 @@
+      - WORDPRESS_DB_USER=wordpress
+      image: wordpress:4.7.4
+      port: 80
+version: v1
```
This should be the first line.

malud replied:
Humanised yes, but its still generated output and parseable from sloppy/cli

jzaefferer replied:
Okay, but let's fix that anyway.

jzaefferer commented on pkg/converter/testdata/docker-compose-v3-simple.yml:
```diff
@@ -0,0 +1,57 @@
+   entrypoint: ["/code/entrypoint.sh", "-foo", "bar"]
+   deploy:
+     mode: replicated
+     replicas: 3
```
Shouldn't this and the memory properties below get mapped to `instances` and `mem`?

May need a 2nd test case (non-simple) to be able to compare this to the v2 input.

malud replied:
Yes, but since there is no decision handling this from v2 it gets currently ignored. Can be refined later on.

jzaefferer replied:
Could put that into a separate issue as well.

malud replied:
Resource implementation: https://github.com/sloppyio/sloppose/issues/3

jzaefferer commented on pkg/converter/linker_test.go:
```diff
@@ -9,76 +9,70 @@ import (
+		"BAR":      {"foo:80", true},
+		"SHORT":    {"s:80", true},
+		"FOO_BAR":  {"another.foo:443", true},
+		"FOO_HOST": {"bar", true},
```
Requiring the string 'HOST' in the env var key is a pretty harsh limitation to. Took me a while to understand that based on the implementation. Need to at least add this to the readme, maybe a "Limitations" section.

malud replied:
I have created an issue for this: https://github.com/sloppyio/sloppose/issues/2

jzaefferer reviewed (changes requested):
Mostly reviewing tests...

Commit:
added missing usage descriptions

Commit:
fixed non working sloppy compose-example

updated libcompose dependency due to global volume related fixes

Commit:
added missing mapping for the logging field
